Formatting nits (#4)
raphaelrobert/privacypass-batched-tokens: draft-robert-privacypass-batched-tokens.md

to be used with the PrivateToken HTTP authentication scheme defined in
{{!AUTHSCHEME=I-D.ietf-privacypass-auth-scheme}}.

# Motivation

Privately Verifiable Tokens (as defines in
{{!ISSUANCE=I-D.ietf-privacypass-protocol}}) offer a simple way to unlink the
issuance from the redemption. The base protocol however only allows for a single
token to be issued at a time for every challenge. In some cases, especially
where a large number of clients need to fetch a large number of tokens, this may
introduce performance bottlenecks. The Batched Token Issuance Protocol improves
upon the basic Privately Verifiable Token issuance protocol in the following key
ways:

1. Issuing multiple tokens at once in response to a single TokenChallenge,
   thereby reducing the size of the proofs required for multiple tokens.
1. Improving server and client issuance efficiency by amortizing the cost of the
   VOPRF proof generation and verification, respectively.

# Client-to-Issuer Request {#client-to-issuer-request}

Except where specified otherwise, the client follows the same protocol as
described in {{ISSUANCE, Section 5.1}}.

The Client first creates a context as follows:

~~~

> client_context = SetupVOPRFClient(0x0001, pkI)

Reviewer: ```suggestion
client_context = SetupVOPRFClient("ristretto255-SHA512", pkI)
```

~~~

> Here, 0x0001 is the two-octet identifier corresponding to the OPRF(ristretto255,

Reviewer: ```suggestion
Here, "ristretto255-SHA512" is the identifier corresponding to the OPRF(ristretto255,
```

- The TokenRequest contains a supported token_type equal to 0xF91A.
- The TokenRequest.token_key_id corresponds to a key ID of a Public Key owned by
  the issuer.
- Nr, as determined based on the size of TokenRequest.blinded_elements, is
  less than or equal to the number of tokens that the issuer can issue in a
  single batch.

If any of these conditions is not met, the Issuer MUST return an HTTP 400 error
to the client.

# Issuer-to-Client Response {#issuer-to-client-response}

Except where specified otherwise, the client follows the same protocol as
described in {{ISSUANCE, Section 5.2}}.

Upon receipt of a TokenRequest, the Issuer tries to deseralize the i-th element
of TokenRequest.blinded_elements using DeserializeElement from {{Section 2.1 of
OPRF}}, yielding `blinded_element_i` of type `Element`. If this fails for any of
the TokenRequest.blinded_elements values, the Issuer MUST return an HTTP 400
error to the client. Otherwise, if the Issuer is willing to produce a token to
the Client, the issuer forms a list of `Element` values, denoted
`blinded_elements`, and computes a blinded response as follows:

~~~

> server_context = SetupVOPRFServer(0x0001, skI, pkI)

Reviewer: ```suggestion
server_context = SetupVOPRFServer("ristretto255-SHA512", skI, pkI)
```
